packages/compass-sidebar/src/components/connections-filter-popover.tsx

```diff
@@ -0,0 +1,125 @@
+import React, { useCallback, type PropsWithChildren } from 'react';
+
+import {
+  css,
+  Icon,
+  IconButton,
+  InteractivePopover,
+  Label,
+  Overline,
+  palette,
+  spacing,
+  Toggle,
+  Tooltip,
+  useId,
+} from '@mongodb-js/compass-components';
+import type { ConnectionsFilter } from './use-filtered-connections';
+
+const containerStyles = css({
+  display: 'flex',
+  flexDirection: 'column',
+  alignItems: 'flex-start',
+  padding: spacing[300],
+  minWidth: 270,
+});
+
+const activatedIndicatorStyles = css({
+  position: 'absolute',
+  top: spacing[50],
+  right: spacing[50],
+});
+
+const groupStyles = css({
+  display: 'flex',
+  flexDirection: 'row',
+  gap: spacing[200],
+  marginTop: spacing[200],
+});
+
+type ConnectionsFilterPopoverProps = PropsWithChildren<{
+  open: boolean;
+  setOpen: (open: boolean) => void;
+  filter: ConnectionsFilter;
+  onFilterChange(
+    updater: (filter: ConnectionsFilter) => ConnectionsFilter
+  ): void;
+}>;
+
+export default function ConnectionsFilterPopover({
+  open,
+  setOpen,
+  filter,
+  onFilterChange,
+}: ConnectionsFilterPopoverProps) {
+  const onExcludeInactiveChange = useCallback(
+    (excludeInactive: boolean) => {
+      onFilterChange((filter) => ({
+        ...filter,
+        excludeInactive,
+      }));
+    },
+    [onFilterChange]
+  );
+
+  const excludeInactiveToggleId = useId();
+  const excludeInactiveLabelId = useId();
+
+  // Add future filters to the boolean below
+  const isActivated = filter.excludeInactive;
+
+  return (
+    <InteractivePopover
+      open={open}
+      setOpen={setOpen}
+      blurTriggerOnClose
+      containerClassName={containerStyles}
+      hideCloseButton
+      trigger={({ onClick, children, ref }) => (
+        <>
+          <Tooltip
+            align="right"
+            trigger={
+              <IconButton
+                onClick={onClick}
+                active={open}
+                aria-label="Filter connections"
+                ref={ref as React.Ref<unknown>}
```

Comment:
To be honest, I didn't fully understand the need for this. It seems the `InteractivePopover` is using the `React.LegacyRef` type for `ref`, but 🤷 it works.

Reply:
InteractivePopover was initially built to be used with `button` which has this specific type of the ref. It's our code, so if we need to change this to Ref (can probably change it to be a more generic HTMLElement instead of HTMLButtonElement too) in the component types, we can, it would make sense and would still be compatible with types of "normal" html button refs

Reply:
If you're okay with it, I suggest making that a follow-up PR.